packages/headless/src/controllers/insight/facets/range-facet/numeric-facet/headless-insight-numeric-filter.ts

```diff
@@ -0,0 +1,92 @@
+import {
+  ConfigurationSection,
+  NumericFacetSection,
+  SearchSection,
+} from '../../../../../state/state-sections';
+import {loadReducerError} from '../../../../../utils/errors';
+import {
+  configuration,
+  numericFacetSet,
+  search,
+} from '../../../../../app/reducers';
+
+import {
+  NumericFilterOptions,
+  NumericFilterInitialState,
+  NumericFilterRange,
+  NumericFilterProps,
+  NumericFilterState,
+  NumericFilter,
+  buildCoreNumericFilter,
+} from '../../../../core/facets/range-facet/numeric-facet/headless-core-numeric-filter';
+import {InsightEngine} from '../../../../../app/insight-engine/insight-engine';
+import {
+  logFacetClearAll,
+  logFacetSelect,
+} from '../../../../../features/facets/facet-set/facet-set-insight-analytics-actions';
+import {executeSearch} from '../../../../../features/insight-search/insight-search-actions';
+
+export type {
+  NumericFilterOptions,
+  NumericFilterInitialState,
+  NumericFilterRange,
+  NumericFilterProps,
+  NumericFilterState,
+  NumericFilter,
+};
+
+/**
+ * Creates a `NumericFilter` controller instance.
+ * @param engine - The headless engine.
+ * @param props - The configurable `NumericFilter` controller properties.
+ * @returns A `NumericFilter` controller instance.
+ */
+export function buildInsightNumericFilter(
```

Comment:
The product listing facet is built with `buildNumericFacet`. Perhaps we should do the same for insight?

Reply:
I was going to alias the function names at export time (in `insight.index.ts`), but since:
1. we need controller builder functions to be the same across use cases
2. and there is no reason to export 2 builder functions for the same controller in a given use case

it would be simpler to use the same name from the start as you suggest.